File: StreamingApp/ClientApp/src/app/api/dtos/search-result-dto.ts
import { ApplicationUserDto } from "./application-user-dto";
import { PlaylistBriefDto } from "./playlist-brief-dto";
import { SongDto } from "./song-dto";

export interface SearchResultDto {
    songs: Array<SongDto>;
    playlistBriefs?: Array<PlaylistBriefDto>;
    users?: Array<ApplicationUserDto>;
}

File: StreamingApp.Domain/DTOs/SearchResultDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StreamingApp.Domain.DTOs
{
    public class SearchResultDto
    {
        public List<SongDto> Songs { get; set; }

        public List<PlaylistBriefDto> PlaylistBriefs { get; set; }

        public List<ApplicationUserDto> Users { get; set; }
    }
}
